src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -204,8 +204,21 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
                 .Select(task => task.ContinueWith(LogError, TaskContinuationOptions.OnlyOnFaulted))
                 .ToArray();
 
-            var completed = (await Task.WhenAll(tasks))
-                .Where(m => m != null);
+            IEnumerable<IPackageSearchMetadata> completed = null;
+            try
+            {
+                completed = (await Task.WhenAll(tasks))
```

Comment:
If the idea is to ignore errors here then I'd suggest creating "safe" individual tasks
```cs
try
{
  return r.GetPackageMetadataAsync(identity, includePrerelease, cancellationToken);
}
catch
{
  return null;
}```